internal/proto/volumegroup.proto

```diff
@@ -0,0 +1,182 @@
+syntax = "proto3";
+package proto;
+
+option go_package = "github.com/csi-addons/kubernetes-csi-addons/internal/proto";
+
+// VolumeGroup holds the RPC methods for volumeGroup and all the methods it
+// exposes should be idempotent.
+service VolumeGroupController {
+  // CreateVolumeGroup RPC call to create a volume group.
+  rpc CreateVolumeGroup(CreateVolumeGroupRequest)
+      returns (CreateVolumeGroupResponse) {}
+  // ModifyVolumeGroupMembership RPC call to modify a volume group.
+  rpc ModifyVolumeGroupMembership(ModifyVolumeGroupMembershipRequest)
+      returns (ModifyVolumeGroupMembershipResponse) {}
+  // DeleteVolumeGroup RPC call to delete a volume group.
+  rpc DeleteVolumeGroup(DeleteVolumeGroupRequest)
+      returns (DeleteVolumeGroupResponse) {}
+  // ListVolumeGroups RPC call to list volume groups.
+  rpc ListVolumeGroups(ListVolumeGroupsRequest)
+      returns (ListVolumeGroupsResponse) {}
+  // CreateVolumeGroup RPC call to get a volume group.
+  rpc ControllerGetVolumeGroup(ControllerGetVolumeGroupRequest)
+      returns (ControllerGetVolumeGroupResponse) {}
+}
+// CreateVolumeGroupRequest holds the required information to
+// create a volume group
+message CreateVolumeGroupRequest {
+  // suggested name for volume group (required for idempotency)
+  // This field is REQUIRED.
+  string name = 1;
+
+  // params passed to the plugin to create the volume group.
+  // This field is OPTIONAL.
+  map<string, string> parameters = 2;
+
+  // Secrets required by the driver to complete the
+  // create request.
+  string secret_name = 3;
+  string secret_namespace = 4;
+
+  // Specify volume_ids that will be added to the volume group.
+  // This field is OPTIONAL.
+  repeated string volume_ids = 5;
+}
+// CreateVolumeGroupResponse holds the information to send when
+// volumeGroup is successfully created.
+message CreateVolumeGroupResponse {
+  // Contains the identifier of the newly created volume group.
+  // This field is REQUIRED.
+  VolumeGroup volume_group = 1;
+}
+// Information about a specific volumeGroup.
+message VolumeGroup {
```

Comment:
Is there a reason to not use the CSI-Addons VolumeGroup type? Duplicating it looks unneeded to me.

Reply:
We would have to tweak the Volumes type to use in CSI-Addons, we can't use the CSI volume type. That's the reaspn I updated it.

Reply:
Done